### ISSUE TITLE
Sphinx 1.2b1: linkcheck fails on references to Python built-in modules, claiming missing anchor

### ISSUE THREAD

**Issue by shimizukawa** [sphinx-doc/sphinx#1159]
It seems Sphinx 1.2b1 (unlike 1.1.3) does not recognize objects.inv properly on docs.python.org. For instance, creating a simple document that refers to :mod:`warnings`, fails linkcheck claiming "Anchor 'warnings' not found" in http://docs.python.org/library/warnings.html.

objects.inv in that case refers to an anchor named "module-warnings", which appears to not be examined properly in this version of Sphinx/intersphinx.

---
- Bitbucket: https://bitbucket.org/birkenfeld/sphinx/issue/1159
- Originally reported by: [vmalloc](https://bitbucket.org/vmalloc)
- Originally created at: 2013-05-01T09:21:01.185

**Comment by shimizukawa** [comment]
_From WAKAYAMA shirou on 2013-09-09 09:09:18+00:00_

This is caused by objects.inv of the docs.python.org includes two types of entries.
1. http://docs.python.org/3.2/library/warnings.html#module-warnings
2. http://docs.python.org/3.2/library/warnings.html#warnings

Since these are same objtype(py:module) and name(warnings), later one overrides the inventory dict.

This quick patch fixes it.

```
diff -r ef3092d458cc sphinx/ext/intersphinx.py
--- a/sphinx/ext/intersphinx.py Sat Aug 10 15:36:05 2013 +0200
+++ b/sphinx/ext/intersphinx.py Mon Sep 09 18:00:25 2013 +0900
@@ -106,6 +106,8 @@
         if not m:
             continue
         name, type, prio, location, dispname = m.groups()
+        if type in invdata and name in invdata[type]:  # avoid duplicates
+            continue
         if location.endswith(u'$'):
             location = location[:-1] + name
         location = join(uri, location)
```

However, I don't know why this is occured (v1 and v2 compatibility ?) and this patch is correct.

anyone knows?

**Comment by shimizukawa** [comment]
_From Georg Brandl on 2013-09-16 05:54:07+00:00_

Closes #1159: do not emit duplicate inventory entries for py:module objects.

Also add a workaround on the consumer side in intersphinx, so that links
to projects using old versions of Sphinx generate correct links.

→ <<cset ff335454577cb5f4603b899cb140c83ffbfd7593>>

**Comment by shimizukawa** [comment]
_From WAKAYAMA shirou on 2013-09-16 05:57:52+00:00_

Thank you!
